src/webots/nodes/utils/WbSupervisorUtilities.cpp

```diff
@@ -1726,6 +1726,7 @@ void WbSupervisorUtilities::writeConfigure(QDataStream &stream) {
   stream << (short unsigned int)0;
   stream << (unsigned char)C_CONFIGURE;
   stream << (int)selfNode->uniqueId();
+  stream << (int)selfNode->parentNode()->uniqueId();
```

Comment:
Note that if you change the communication messages between Webots and the controller, then this has to be merged on the develop branch and not on the master branch otherwise you will break the compatibility (unless this is a bug fix).

Reply:
I see. Initially, I wanted to change only the parser, but then I figured out it is needed to update the controller as well.

I will make a new PR, so I don't mess up the history.